spectrochempy/analysis/_base.py

```diff
@@ -1043,6 +1085,433 @@ def Y(self):
+            True values for `X`.
+        sample_weight: `NDDataset` or array-like of shape (n_samples,), default=None
+            Sample weights.
+
+        Returns
+        -------
+        score: float
+            :math:`R^2` of `self.predict(X)` w.r.t `Y`
+        """
+        if not self._fitted:
+            raise NotFittedError()
+
+        if X is None:
+            X = self._X_preprocessed
+        elif isinstance(X, NDDataset):
+            X = X.data
+
+        if Y is None:
+            Y = self._Y_preprocessed
+        elif isinstance(Y, NDDataset):
+            Y = Y.data
+
+        if isinstance(sample_weight, NDDataset):
+            sample_weight = sample_weight.data
+
+        return self._score(X, Y, sample_weight)
+
+    @_wrap_ndarray_output_to_nddataset(
+        units=None, title=None, meta_from=("_X", "_Y"), typex="components"
+    )
+    @_docstring.dedent
+    def transform(self, X=None, Y=None, both=False, copy=True, **kwargs):
+        """
+        Apply dimensionality reduction to `X`\ and `Y`\.
+
+        Parameters
+        ----------
+        X : `NDDataset` or :term:`array-like` of shape (:term:`n_observations`\ , :term:`n_features`\ ), optional
+            New data, where :term:`n_observations` is the number of observations
+            and :term:`n_features` is the number of features.
+            if not provided, the input dataset of the `fit` method will be used.
+        Y : `NDDataset` or :term:`array-like` of shape (:term:`n_observations`\ , :term:`n_features`\ ), optional
+            New data, where :term:`n_observations` is the number of observations
+            and :term:`n_features` is the number of features.
+            if not provided, the input dataset of the `fit` method will be used.
+        both: bool, default=False
+            whether to also apply the dimensionality reduction to Y when neither X nor Y are provided
+        copy: bool, default=True
+            Whether to copy X and Y, or perform in-place normalization.
+        %(kwargs)s
+
+        Returns
+        -------
+        x_score, y_score: `~spectrochempy.core.dataset.nddataset.NDDataset` or tuple of
+        `~spectrochempy.core.dataset.nddataset.NDDataset` `
+            Datasets with shape (:term:`n_observations`\ , :term:`n_components`\ ).
+
+        """
+        if not self._fitted:
+            raise NotFittedError()
+
+        # Fire the validation and preprocessing
+        self._X = X if X is not None else self.X
+        self._Y = Y if Y is not None else self.Y
+
+        # Get the processed ndarray data
+        newX = self._X_preprocessed
+        newY = self._Y_preprocessed
+
+        if both or (Y is not None):
+            return self._transform(newX, newY, copy)
+        else:
+            return self._transform(newX, None, copy)
+
+    # Get doc sections for reuse in subclass
+    _docstring.get_sections(
+        _docstring.dedent(transform.__doc__),
+        base="analysis_transform",
```

Comment:
This is the reason for the error in EFA docstring. 
You cannot use the same base (as for DecompositionAnalysis line 809) because you override the previous definition. 
Use a different base for CrossDecomposition, for example=.  "cross_analysis_transform" or whatever you want but not 
' analysis_transform`!